### DISCUSSION TITLE
NITF viewer with nitro

### DISCUSSION THREAD

**Discussion by s2000coder** [mdaus/nitro#328] [Q&A] [answered]
Hello,

I am looking to write python program to display NITF image (i.e., GUI NITF viewer). Can this be done with the nitro library?

Thanks!

**Comment by bradh** [comment]
Yes it can. You can get a PIL Image, as shown in https://github.com/mdaus/nitro/blob/62ec49119354f0001b7575a2e42d9545d6fc1d86/modules/python/nitf/samples/test_image.py#L53

Note that not all NITF files are intended to be displayed. Consult MIL-STD-2500C CN2 for more guidance.

**Comment by s2000coder** [comment]
I am getting the following error when trying to load the PIL module. According to the ctest, nitro was compiled and installed successfully. Any idea?

  ```
File "test_image.py", line 29, in <module>
    from PIL import Image
ModuleNotFoundError: No module named 'PIL'
```

**Comment by bradh** [comment, marked as answer]
I think you didn't install Pillow (or something else with the PIL API). See https://pillow.readthedocs.io/en/latest/installation.html

**Comment by s2000coder** [comment]
@bradh, thank you for your pointers thus far.

I ran into issues with several NITF images. I tested these images on other NITF programs and they displayed the images without any problem, but with NITRO, I got the following error:

```
Traceback (most recent call last):
  File "./other/test_image.py", line 80, in <module>
    dump_images(arg)
  File "./other/test_image.py", line 72, in dump_images
    imReader = reader.newImageReader(i)
  File "/usr/local/lib/python3.6/site-packages/nitf/__init__.py", line 357, in newImageReader
    if not reader: raise Exception('Unable to get new ImageReader')
Exception: Unable to get new ImageReader
```

Both the `record.getImages()` and `record.getTexts()` methods return the same exception, so I am guessing it may be a compression compatibility issue. I notice that any NITF with bi-level image compression runs into this error. Is there a way to get around this?

Thanks!

**Reply by bradh** [reply]
I don't know. Bi-level isn't very common (VQ and JPEG 2000 are much more likely). Can you share a sample file, or at least the image segment header?

**Reply by s2000coder** [reply]
The file I used is number 17 on the list: https://www.gwg.nga.mil/ntb/baseline/software/testfile/Nitfv2_1/scen_2_1.html

**Reply by bradh** [reply]
I wouldn't worry about that specific compression type. It's possible to implement it if you want, but unlikely to be any real-world use of it.
